src/mlpack/methods/ann/dists/bernoulli_distribution.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file bernoulli_distribution.hpp
+ * @author Atharva Khandait
+ *
+ * Definition of the Bernoulli distribution class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_DISTRIBUTIONS_BERNOULLI_DISTRIBUTION_HPP
+#define MLPACK_METHODS_ANN_DISTRIBUTIONS_BERNOULLI_DISTRIBUTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "../activation_functions/logistic_function.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Multiple Bernoulli distributions.
+ *
+ * @tparam DataType Type of the input data. (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <typename DataType = arma::mat>
+class BernoulliDistribution
+{
+ public:
+  /**
+   * Default constructor, which creates a Bernoulli distribution with zero
+   * dimension.
+   */
+  BernoulliDistribution();
+
+  /**
+   * Create multiple Bernoulli distributions with the given parameters.
+   * If the param matrix is already between [0, 1], then applyLogistic can be
+   * set to false.
+   *
+   * @param param The matrix of probabilities or pre probabilities of
```

Comment:
I think 'logits' is more understandable and common than 'pre probabilities'. Also we should mention the how the dimensions of param affect the distribution.